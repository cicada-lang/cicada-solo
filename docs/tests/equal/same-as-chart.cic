same_as_chart! Nat [
  add1(add1(add1(0))),
  add1(add1(1)),
  add1(2),
  3,
]

_: Equal(Nat, add1(add1(add1(0))), 3) {
  _ = is(same(add1(add1(add1(0)))), Equal(Nat, add1(add1(add1(0))), add1(add1(1))))
  _ = is(same(add1(add1(add1(0)))), Equal(Nat, add1(add1(1)), add1(2)))
  _ = is(same(add1(add1(add1(0)))), Equal(Nat, add1(2), 3))
  _
}

_
