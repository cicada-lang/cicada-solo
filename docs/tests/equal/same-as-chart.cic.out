refl: Equal(String, "abc", "abc")
refl: Equal(String, "abc", "abc")
refl: Equal(String, "abc", "abc")